packages/Tables/index.tsx

```diff
@@ -0,0 +1,161 @@
+import * as React from 'react';
+
+import {
+  ScrollView,
+  StyleProp,
+  StyleSheet,
+  Text,
+  View,
+  ViewStyle,
+} from 'react-native';
+
+import CheckBox from './CheckBox';
+import TableCell from './TableCell';
+import TableHeader from './TableHeader';
+import TableRow from './TableRow';
+import TableTitle from './TableTitle';
+import styled from 'styled-components/native';
+
+export interface Props {
+  style?: StyleProp<ViewStyle>;
+  data: Array<Record<string, any>>;
+  isCheckAble?: boolean;
+  customGroup?: Array<string>;
+}
+
+type TableNamespace = {
+  Title: typeof TableTitle;
+  Header: typeof TableHeader;
+  Row: typeof TableRow;
+  Cell: typeof TableCell;
+};
+
+const Container = styled.View`
+  flex: 1;
+  border-top-left-radius: 4px;
+  border-top-right-radius: 4px;
+`;
+
+const Table: React.FC<Props> & TableNamespace = ({
+  data,
+  isCheckAble,
+  customGroup,
+  style,
+  ...rest
+}) => {
+  /** default Data key */
+  const group = data.reduce((acc, current) => {
+    Array.from(Object.assign(acc, Object.keys(current)));
+    return acc;
+  }, []);
+
+  /** checking interaction */
+  const [selected, setSelected] = React.useState<string[]>([]);
+  const isSelected = (name: string): boolean => selected.indexOf(name) !== -1;
+  const handleClick = (name: string): void => {
+    const selectedIndex = selected.indexOf(name);
+    let newSelected: string[] = [];
+    if (selectedIndex === -1) {
+      newSelected = newSelected.concat(selected, name);
+    } else if (selectedIndex === 0) {
+      newSelected = newSelected.concat(selected.slice(1));
+    } else if (selectedIndex === selected.length - 1) {
+      newSelected = newSelected.concat(selected.slice(0, -1));
+    } else if (selectedIndex > 0) {
+      newSelected = newSelected.concat(
+        selected.slice(0, selectedIndex),
+        selected.slice(selectedIndex + 1),
+      );
+    }
+    setSelected(newSelected);
+  };
+  /** Render */
+  return (
+    <ScrollView
+      style={{
+        width: '100%',
+        height: '100%',
+      }}>
+      <ScrollView horizontal>
+        <Container {...rest} style={[style]}>
```

Comment:
Is this important to expose all the props if view here? It'd be better to provide more specific props that you think is necessary.